plugins/org.eclipse.fordiac.ide.typemanagement/src/org/eclipse/fordiac/ide/typemanagement/refactoring/ConnectionsToStructCommand.java

```diff
@@ -0,0 +1,276 @@
+	}
+
+	private void connectStruct() {
+		connectStructCommand = new CompoundCommand();
+
+		// Create map between correct connected FBs and create MUX
+		getElementsOfType(destinationType).stream().forEach(instance -> {
+			instance.getInterface().getErrorMarker().stream().flatMap(err -> err.getInputConnections().stream());
+
+			// Collect all correct connections
+			final List<Connection> cons = instance.getInterface().getErrorMarker().stream()
+					.flatMap(err -> err.getInputConnections().stream())
+					.filter(con -> replacableConMap.containsKey(con.getSource().getName())
+							&& replacableConMap.get(con.getSource().getName()).equals(con.getDestination().getName())
+							&& con.getSourceElement().getType().getName().equals(sourceType.getName()))
+					.toList();
+
+			// Check if all connections between 2 instances are correct -> store in map
+			if (cons.stream()
+					.map((Function<? super Connection, ? extends FBNetworkElement>) Connection::getSourceElement)
+					.distinct().count() == 1 && cons.size() == replacableConMap.size()) {
+
+				final FBNetworkElement source = instance.getInterface().getErrorMarker().stream()
+						.flatMap(err -> err.getInputConnections().stream())
+						.filter(con -> replacableConMap.containsValue(con.getDestination().getName())).findFirst().get()
+						.getSourceElement();
+				final StructDataConnectionCreateCommand structCon = new StructDataConnectionCreateCommand(
+						instance.getFbNetwork());
+				structCon.setDestination(instance.getInput(destinationVarName));
+				structCon.setSource(source.getOutput(sourceVarName));
+				connectStructCommand.add(structCon);
+				instance.getInterface().getErrorMarker().stream().flatMap(err -> err.getInputConnections().stream())
+						.filter(con -> replacableConMap.containsValue(con.getDestination().getName()))
+						.forEach(con -> connectStructCommand.add(new DeleteConnectionCommand(con)));
+			}
+		});
+		connectStructCommand.execute();
+	}
+
+	private void conflictResolution() {
+		destinationConnectionEditCommand = new CompoundCommand();
+		sourceConnectionEditCommand = new CompoundCommand();
+
+		// Edit destination connections
+		getElementsOfType(destinationType).stream().forEach(instance -> {
+			instance.getInterface().getErrorMarker().stream().flatMap(err -> err.getInputConnections().stream())
+					.filter(con -> replacableConMap.containsValue(con.getDestination().getName())).forEach(con -> {
+						final String var = replacableConMap.entrySet().stream()
+								.filter(entry -> entry.getValue().equals(con.getDestination().getName())).findAny()
+								.get().getKey();
+						final RepairBrokenConnectionCommand rep = new RepairBrokenConnectionCommand(con, false,
+								(StructuredType) structType, var);
+						destinationConnectionEditCommand.add(rep);
+					});
+		});
+		destinationConnectionEditCommand.execute();
+
+		// Edit source connections
+		getElementsOfType(sourceType).stream().forEach(instance -> {
```

Comment:
instead of commenting you can make a method editConnections(boolean isSource,...) . so you might also replace the code from 203-212 . It looks quite similar